packages/nimble-components/src/combobox/index.ts

```diff
@@ -68,72 +106,234 @@ export class Combobox
 
     /** @internal */
     @observable
     public hasOverflow = false;
 
     public override get value(): string {
-        return super.value;
+        Observable.track(this, 'value');
+        return this._value;
     }
 
-    // This override is to work around an issue in FAST where an old filter value
-    // is used after programmatically setting the value property.
-    // See: https://github.com/microsoft/fast/issues/6749
     public override set value(next: string) {
-        super.value = next;
-        // Workaround using index notation to manipulate private member
+        const prev = `${this._value}`;
+        let updatedValue = next;
+
+        if (this.$fastController.isConnected && this.options) {
+            const selectedIndex = this.options.findIndex(
+                el => el.text.toLowerCase() === next.toLowerCase()
+            );
+
+            const prevSelectedValue = this.options[this.selectedIndex]?.text;
+            const nextSelectedValue = this.options[selectedIndex]?.text;
+
+            this.selectedIndex = prevSelectedValue !== nextSelectedValue
+                ? selectedIndex
+                : this.selectedIndex;
+
+            updatedValue = this.firstSelectedOption?.text || next;
+        }
+
+        if (prev !== updatedValue) {
+            this._value = updatedValue;
+            super.valueChanged(prev, updatedValue);
+            Observable.notify(this, 'value');
+        }
+
         // Can remove when following resolved: https://github.com/microsoft/fast/issues/6749
-        // eslint-disable-next-line @typescript-eslint/dot-notation
-        this['filter'] = next;
+        this.filter = next;
         this.filterOptions();
         this.selectedIndex = this.options
             .map(option => option.text)
             .indexOf(this.value);
     }
 
+    /**
+     * The list of options.
+     *
+     * Overrides `Listbox.options`.
+     */
+    public override get options(): ListboxOption[] {
+        Observable.track(this, 'options');
+        return this.filteredOptions?.length
+            ? this.filteredOptions
+            : this._options;
+    }
+
+    public override set options(value: ListboxOption[]) {
+        this._options = value;
+        Observable.notify(this, 'options');
+    }
+
+    /**
+     * The unique id for the internal listbox element.
+     *
+     * @internal
+     */
+    public listboxId: string = uniqueId('listbox-');
+
+    /**
+     * The max height for the listbox when opened.
+     *
+     * @internal
+     */
+    @observable
+    public maxHeight = 0;
+
     private valueUpdatedByInput = false;
     private valueBeforeTextUpdate?: string;
+    private _value = '';
+    private filter = '';
+
+    /**
+     * The initial state of the position attribute.
+     */
+    private forcedPosition = false;
 
-    // Workaround for https://github.com/microsoft/fast/issues/5123
-    public override setPositioning(): void {
+    public setPositioning(): void {
```

Comment:
this should be internal right? weird that it's not in fast..

Reply:
Yeah I only didn't mark it that way, because it wasn't originally in FAST. I'll change.